tests/framework/e2e/cluster.go

```diff
@@ -813,6 +821,33 @@ func (epc *EtcdProcessCluster) StartNewProc(ctx context.Context, cfg *EtcdProces
 	return proc.Start(ctx)
 }
 
+// UpdateProcOptions updates the options for a specific process. If no opt is set, then the config is identical
+// to the cluster.
+func (epc *EtcdProcessCluster) UpdateProcOptions(i int, tb testing.TB, opts ...EPClusterOption) error {
+	cfg := *epc.Cfg
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+	serverCfg := cfg.EtcdServerProcessConfig(tb, i)
+
+	var initialCluster []string
+	for _, p := range epc.Procs {
+		initialCluster = append(initialCluster, fmt.Sprintf("%s=%s", p.Config().Name, p.Config().PeerURL.String()))
+	}
+	epc.Cfg.SetInitialOrDiscovery(serverCfg, initialCluster, "new")
+
+	proc, err := NewEtcdProcess(serverCfg)
+	if err != nil {
+		return err
+	}
+	err = epc.Procs[i].Close()
+	if err != nil {
+		return err
+	}
```

Comment:
UpdateProcOptions function stopping a process is a little unexpected for me. I know that to override the `epc.Procs[i] = proc` you need to stop old process, but the approach here is not very natural. 

What about `UpdateProcOptions` expecting that `epc.Procs[i]` is already stopped and assert that? Before calling `UpdateProcOptions` test should close the process.